### PR TITLE
Void type not allowed in __arglist

### PR DESCRIPTION
**Pull request by jcouv**
`__arglist(VoidReturningMethod())` should produce an error. It necessarily fails at runtime.

Fixes https://github.com/dotnet/roslyn/issues/23369

### Risk & Performance impact
Low. Just adding a simple check to `__arglist` arguments.

### Is this a regression from a previous update?
No

### How was the bug found?
Reshetnikov

### CODE REVIEW

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler Adding a diagnostic.
In terms of compat break, I will confirm with the compat council, but such code fails at runtime anyways.

**Comment by jcouv** [comment]
@dotnet/roslyn-compat-council is ok with adding this diagnostic.

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for review. Tiny fix (adding a diagnostic). Thanks

File: src/Compilers/CSharp/Portable/Errors/ErrorCode.cs [hunk 1]
```diff
@@ -1539,7 +1539,6 @@ internal enum ErrorCode
         ERR_OutAttrOnInParam = 8355,
         #endregion diagnostics introduced for `ref readonly`, `ref ternary` and `ref-like` features in C# 7.2
 
-        #region diagnostics introduced for C# 7.3
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Errors/ErrorCode.cs]: Was this change intentional?

Comment (jcouv) [reply src/Compilers/CSharp/Portable/Errors/ErrorCode.cs]: Yes, this is intentional.
I'm not sure why this comment landed here. But there is no LanguageVersion for C# 7.3 in master ([see](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/LanguageVersion.cs#L121)).

Comment (jcouv) [reply src/Compilers/CSharp/Portable/Errors/ErrorCode.cs]: Work for C# 7.3 is happening in the `features/compiler` branch (which may align with15.7).

Comment (jcouv) [reply src/Compilers/CSharp/Portable/Errors/ErrorCode.cs]: @gafter I see that that comment was added in one of your PRs (https://github.com/dotnet/roslyn/pull/23265/files#diff-3ff29d8326aea505287a1f262172c8a3). FYI, I'm removing it.

**Comment by jcouv** [comment]
@gafter @VSadov for a second review. Thanks

**Comment by jcouv** [comment]
Thanks
@jaredpar for ask-mode approval for 15.6.

**Comment by jcouv** [comment]
@MeiChin-Tsai Please approve for ask-mode for 15.6. Thanks

**Comment by MeiChin-Tsai** [comment]
approved
